### PR TITLE
Add initial implementation to provision TwinCAT/BSD PLCs with ansible

### PR DESCRIPTION
**Pull request by klauer**
# Overview

* A VM creation tool
* Ansible playbook for provisioning the new VM

## VM creation tool

Simple script for making a VirtualBox VM from Beckhoff's BSD image

## Ansible playbook: provision the BSD-based PLC

(Disclaimer: my first time using ansible was ~yesterday~ a day before this PR was opened)

* Bootstraps the PLC by installing Python 3.9 (required for ansible)
* Installs system packages that I like and find useful to have everywhere
* Installs system packages specified in the host inventory
* Installs TwinCAT tools and a couple libraries
* Lists all of the available libraries I saw in `pkg search`, with easy uncomment-ability
* Sets an AMS Net ID based on the host inventory
* Sets locked memory size in the TcRegistry based on the host inventory
* (Optionally) Sets heap memory size in the TcRegistry based on the host inventory
* (Optionally) Installs C/C++ development tools
* (Optionally) Changes the default shell for Administrator to `bash`
* (Optionally) Configures initial static routes, by default just adding the
  machine from which ansible is being run.
* (Optionally) Configures add or remove a list of static routes with a custom
  ansible module ``tcbsd_route`` (see ``library/``).
* Provides a basic bash configuration, which shows the current TC runtime state:
  ```
  [TCBSD: CONFIG] [Administrator@PC-75972A  ~]$
  [TCBSD: RUN] [Administrator@PC-75972A  ~]$
  ```
* Enable color in ``ls`` and bash tab completion, if using bash.
* Adds a "site" firewall (pf = packet filter) configuration which lets through insecure ADS
* Reloads the packet filter if the configuration was changed
* Restarts the TwinCAT service if any TcRegistry changes were made

## TODO
- [x] Better document the process - start from scratch again
- [x] Think about additional settings that would be advisable
- [x] Include route additions directly from the PLC with the ads tool?
- [x] Ensure PMPS database uploading has a non-FTP path forward
- [ ] (Optional) LCLS-specific NTP settings

### CODE REVIEW

**Comment by klauer** [comment]
Pre-installed BSD image on a CX5240-0185 notes:
* DHCP was enabled (good)
* SSH didn't work for initial setup
    * Requires then a monitor and a keyboard
    * Had to enable password-based auth to remotely manage it
* For our development network, package upgrades behind a proxy (`psproxy`) required additional `pkg` settings to get working
* Pre-installed packages and TwinCAT version were old (*)
* Package upgrades didn't work because the system time was off (and certificates weren't valid for 2020)
* NTP configuration needs to be added here because of this (this should be one of the first bootstrapping steps for us)


(*) Upgrade steps:
```
$ sudo pkg update -f 
(You may have to enter 'y' here)
$ sudo pkg upgrade
```
There was no major system update for the provided version, so the following steps were not required:
```
$ sudo pkg install tcbsd-upgrade
$ sudo tcbsd-upgrade major
```

Other notes:
* Unable to install TC/BSD on a CX5010-0110 as it does not support UEFI boot (maybe a bios update is available?)

**Comment by klauer** [comment]
There's still stuff to do here, but I'd like to merge this and follow up on it. 

This is entirely out of our team's wheelhouse so with the lack of appropriate reviewers, I'm calling on the (unfortunate) usual suspects.

File: .shellcheckrc
Comment (ZLLentz) [inline .shellcheckrc:1]: TIL, the existence of this file makes me want to proliferate the shellcheck pre-commit

**Review by ZLLentz — APPROVED** [review]
I don't have anything useful to contribute here lgtm

**Review by tangkong — APPROVED** [review]
I'll stamp this, lest I hold it back.  👍

**Pull request merged**